x-pack/plugins/observability_solution/apm/server/lib/apm_telemetry/collect_data_telemetry/tasks.ts

```diff
@@ -695,10 +692,7 @@ export const tasks: TelemetryTask[] = [
         },
       });
 
-      const hit = normalizeFields(response.hits.hits[0]?.fields) as Pick<
-        Transaction | Span | APMError,
-        'observer'
-      >;
+      const hit = serviceVersionMapping(response.hits.hits[0]?.fields);
```

Comment:
If this returns APMError, special care will need to be taken. I'm not sure that `fields` is capable of returning fields from `error.exception.stacktrace`.  We may have to depend on _source in this situation.

Reply:
Here's an example query for demonstration.
```
GET /_search 
{
        "fields": ["error.exception.message","error.exception.stacktrace.exclude_from_grouping"],
            "query": {
        "exists": {
            "field": "error.exception.message"
        }  
    
    },
    "size": 1
}
```